torch/csrc/distributed/rpc/script_functions.cpp

```diff
@@ -0,0 +1,53 @@
+#include <torch/csrc/distributed/rpc/script_functions.h>
+
+#include <torch/csrc/distributed/autograd/utils.h>
+#include <torch/csrc/distributed/rpc/message.h>
+#include <torch/csrc/distributed/rpc/python_rpc_handler.h>
+#include <torch/csrc/distributed/rpc/rpc_agent.h>
+#include <torch/csrc/distributed/rpc/script_call.h>
+#include <torch/csrc/distributed/rpc/utils.h>
+
+namespace torch {
+namespace distributed {
+namespace rpc {
+
+c10::IValue rpcTorchscriptCall(
+    const std::string& dst,
+    const c10::QualifiedName& qualifiedName,
+    std::vector<c10::IValue>& stack) {
+  auto scriptCall =
+      c10::guts::make_unique<ScriptCall>(qualifiedName, std::move(stack));
+  auto agent = RpcAgent::getDefaultRpcAgent();
+  auto futMessage = autograd::sendMessageWithAutograd(
+      *agent, agent->getWorkerInfo(dst), std::move(*scriptCall).toMessage());
+  // Get function return type to construct c10::ivalue::Future.
+  // Script call only allows single IValue returned.
+  auto returns = PythonRpcHandler::getInstance()
```

Comment:
`ScriptCall` should just record the return type that we get from the compilation unit.